### PR TITLE
ec2 inventory allow for AND usage when specifying multiple tags for instances

### PR DESCRIPTION
**Pull request by kapilt**
pretty common to want the intersection of multiple tags, ie my app tag, and the 'dev' stage in an account with lots of apps/stages.

### CODE REVIEW

**Comment by bcoca** [comment]
@kapilt could squash to a single commit?

**Comment by kapilt** [comment]
@bcoca squashed and minor doc cleanup in config example

File: contrib/inventory/ec2.py [hunk 1]
```diff
@@ -397,6 +397,14 @@ def read_settings(self):
                     continue
                 self.ec2_instance_filters[filter_key].append(filter_value)
 
+        # Instance Filters can be AND'd or OR'd based on the filters_operator
+        # parameter, default behavior is OR'd
+        self.ec2_filter_operator = 'or'
+        if config.has_option('ec2', 'filters_operator'):
+            self.ec2_filter_operator = config.get('ec2', 'filters_operator')
+            if self.ec2_filter_operator not in ('or', 'and'):
+                self.ec2_filter_operator = 'or'
```
Comment (mgedmin) [inline contrib/inventory/ec2.py]: Would be nice to warn the user their configuration is being ignored because the specified value wasn't recognized.

Comment (kapilt) [inline contrib/inventory/ec2.py]: noted, sounds good

**Comment by aioue** [comment]
Related: https://github.com/ansible/ansible/pull/14901

**Comment by kapilt** [comment]
@wtcross here

**Comment by bcoca** [comment]
also linking #10413 as it was another attempt to same using a different approach

**Comment by nicolas-g** [comment]
+1

**Comment by mkrizek** [comment]
Fixed in https://github.com/ansible/ansible/pull/30272

**Pull request closed without merging**